build_package.py

```diff
@@ -107,6 +107,15 @@ def generate_thrift_files(thrift_files_dir, env, silent=True):
         LOG.error('Failed to generate viewer server files')
         return ret
 
+    auth_thrift = os.path.join(thrift_files_dir, 'authentication.thrift')
+    auth_thrift = 'authentication.thrift'
+    auth_cmd = ['thrift', '-r', '-I', '.',
+                '--gen', 'py', auth_thrift]
+    ret = run_cmd(auth_cmd, thrift_files_dir, env, silent=silent)
+    if ret:
+        LOG.error('Failed to generate authentication interface files')
+        return ret
```

Comment:
So far we generated the source files with Thrift files differently for the report server (python only) and the report viewer (python and javascript). The authentication is python only which could be refactored into a separate function with the report server source file generation. I'm not sure it will be so much shorter,  right now the generate_thrift_files function is not that long. If you think it is better you can refactor it into a separate function.